### PR TITLE
feat: mandatory suppression explanations

### LINKED ISSUE #972: 💭 Consider mandatory lint suppression explanations

### PR DESCRIPTION
**Pull request by bitpshr**
## Summary

This pull request makes suppression explanations mandatory.

-   Suppression linting requires explanations.
-   CLI and LSP decisions add placeholder suppression explanations.
-   A new `--suppression-explanation` review CLI flag is supported.
-   Documentation explains that explanations are mandatory.

Resolves #972

## Test Plan

```sh
./rome test
```

### CODE REVIEW

**Comment by ematipico** [comment]
@bitpshr: is this PR changes also how `rome check --review` works? In theory now, when deciding to suppress the rule, we should prompt the reason why that rule must be suppressed.

**Comment by bitpshr** [comment]
@sebmck I worry about opening up something like suppression explanations to configuration since their benefit is fairly objective: including them will always increase suppression understanding, whereas not including them only ever saves keystrokes. I feel that this is similar to implicit `return` statements or any other keystroke-saving-but-technically-bad-practice rule, and I think Rome should lean on objectivity in these types of cases rather than opening them up.

@ematipico `rome check --review` inserts a placeholder suppression explanation for now. We can definitely improve the flow to accept user input instead; I'll have to walk the CLI command code a bit more to properly implement this.

**Comment by sebmck** [comment]
I worry about making them mandatory without an option to make them optional. These serious of "microaggressions" that Rome has I fear will really stunt adoption. I think it's a good balance having it mandatory by default, where we can impose our opinions, while still making it configurable for those opinionated who will likely seek out a configuration option when they encounter it as a blocker.

**Comment by bitpshr** [comment]
Good point about configuration facilitating greater Rome adoption @sebmck. I also think that rules and configuration options can be tightened as trust in the tooling builds across the community.

Configuration option added, ready for another pass.

File: website/src/_includes/docs/project-config.md [hunk 1]
```diff
@@ -75,6 +75,14 @@ Custom variables you want to declare as global.
 rome config push lint.globals SomeGlobal
 ```
 
+#### `lint.requireSuppressionExplanations`
+
+Raise a diagnostic if a suppression does not have a [valid explanation](#explanation).
+
+```bash
+rome config push lint.globals SomeGlobal
```
Comment (sebmck) [inline website/src/_includes/docs/project-config.md:83]:
```suggestion
rome config enable lint.requireSuppressionExplanations
```

**Pull request merged**